src/main/java/liqp/Template.java

```diff
@@ -689,4 +698,67 @@ private void putStringKey(boolean convertValueToMap, String key, Object value, M
             map.put(key, value);
         }
     }
+
+    private static List<String> tokenizeString(String input) {
+        List<String> tokens = new ArrayList<>();
+
+        Pattern pattern = Pattern.compile("\\{\\{.*?\\}\\}|\\s+|\\S+");
+        Matcher matcher = pattern.matcher(input);
+        while (matcher.find()) {
```

Comment:
tried it. since with matcher.results() we collect all the matches together, we got a few issues after I updated it.
so I changed it back to find() because if .results() throws exception with any collection of placeholder, we will catch Exception for the whole string itself.